File: android_app/app/src/test/java/comp4350/triviasmack/tests/objects/QuestionTest.java
package comp4350.triviasmack.tests.objects;

import junit.framework.TestCase;

import comp4350.triviasmack.objects.Question;

import static org.junit.Assert.assertArrayEquals;

public class QuestionTest extends TestCase {

    private String question;
    private String[] options;
    private int answer;
    private Question questionObj;

    public QuestionTest(String arg0) {
        super(arg0);
    }

    @Override
    protected void setUp() throws Exception {
        question = "What's my favorite color?";
        options = new String[]{"blue", "green", "red"};
        answer = 0;
        questionObj = new Question(question, options, answer);
    }

    @Override
    protected void tearDown() throws Exception {
        questionObj = null;
    }

    public void testConstructor() {
        System.out.println("Testing Question: Constructor");

        assertNotNull(questionObj);
        assertEquals(question, questionObj.getQuestion());
        assertArrayEquals(options, questionObj.getOptions());
        assertEquals(answer, questionObj.getAnswer());
    }

    public void testAccessors() {
        System.out.println("Testing Question: Mutators");

        assertEquals(answer, questionObj.getAnswer());
        assertEquals(options, questionObj.getOptions());
        assertEquals(question, questionObj.getQuestion());
    }

    public void testEmptyQuestion() {
        System.out.println("Testing Question: EmptyQuestion");

        Question q1 = new Question("", new String[0], 0);

        assertEquals(q1.getQuestion(), "");
        assertEquals(q1.getOptions().length, 0);
        assertEquals(q1.getAnswer(), 0);
    }

    public void testFailure() {
        System.out.println("Testing Question: Invalid Args");

        try{
            new Question(null, null, -1);
            fail("Expected a NullPointerException");
        } catch (NullPointerException ignored) {}
    }
}

File: web_app/tests/objects/QuestionTest.py
"""
QuestionTest.py
"""
import unittest
from web_app.objects.Question import Question

class QuestionTest(unittest.TestCase):
    """Unit tests for the Question class"""

    def setUp(self):
        self.question = "What's my favorite color?"
        self.options = ["blue", "green", "red"]
        self.answer = 0
        self.question_obj = Question(self.question, self.options, self.answer)

    def tearDown(self):
        self.question_obj = None

    def test_init(self):
        print "Testing Question: Constructor"

        self.assertIsNotNone(self.question_obj)
        self.assertEquals(self.question, self.question_obj.question)
        self.assertEquals(self.options, self.question_obj.options)
        self.assertEquals(self.answer, self.question_obj.answer)

    def test_mutation(self):
        print "Testing Question: Mutators"

        question = "What kind of bear is best?"
        options = ["grizzly", "black", "polar"]
        answer = 1

        self.question_obj.question = question
        self.question_obj.options = options
        self.question_obj.answer = answer

        self.assertEquals(question, self.question_obj.question)
        self.assertEquals(options, self.question_obj.options)
        self.assertEquals(answer, self.question_obj.answer)

    def test_failure(self):
        print "Testing Question: Invalid Args"
        try:
            Question()
            self.fail("Expected typeError")
        except TypeError:
            pass
